### PR TITLE
Port command-related integration tests

### LINKED ISSUE #464: Get rid of the default timeout in `FutureTimeouts.timeout`
See:
https://github.com/digital-asset/daml/blob/master/ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/tests/integration/ledger/api/FutureTimeouts.scala#L12-L16

I believe this class is mostly used for testing.
However we should not rely on default timeout values. Every test has to decide what the appropriate timeout is.
I have tried to get rid of the default, it affects too many thing. Need a separate issue.

CC @francesco-da @gaboraranyossy-da

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Not a full port of any particular suite, this mostly focuses on deleting as much lingering code as possible while porting a few tests from various command-related suites (and removing two or three duplicate test cases already covered by the transaction service new tests).

Also removes code previously put there to make the old integration tests runnable against an external sandbox and flattens the fixture traits in a single place to avoid resource initialization issues.

Closes #464.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
         displayName: nix cache key
       - task: CacheBeta@0
         inputs:
-          key: $(nix-cache-key)
+          key: $(nix-cache-key) | dQw4w9WgXcQ
```
Comment (gerolf-da) [inline azure-pipelines.yml]: @stefanobaghino-da: do I want to know about this?

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala [hunk 2]
```diff
@@ -262,6 +275,282 @@ final class CommandService(session: LedgerSession) extends LedgerTestSuite(sessi
...
+      } yield {
+        val created = assertSingleton("There should only be one creation", createdEvents(tree))
+        assertEquals(
+          "The created event should be the expected one",
+          created.getCreateArguments.fields,
+          encode(CallablePayout(giver, newReceiver)).getRecord.fields)
+      }
+    }
+
+  private[this] val unitAsArgumentToNothing =
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: I don't understand what this test is trying to test. Reading the old test doesn't help either. Where does the `Unit` come into play?

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala [hunk 3]
```diff
@@ -262,6 +275,282 @@ final class CommandService(session: LedgerSession) extends LedgerTestSuite(sessi
...
+        tree <- ledger.exercise(party, factory.exerciseDummyFactoryCall)
+      } yield {
+        val exercise = assertSingleton("There should only be one exercise", exercisedEvents(tree))
+        assert(exercise.contractId == factory.unwrap, "Contract identifier mismatch")
+        assert(exercise.consuming, "The choice should have been consuming")
+        val _ = assertLength("Two creations should have occured", 2, createdEvents(tree))
+      }
+    }
+
+  private[this] val completions =
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: Very strictly speaking this test doesn't test the `CommandService` but the `CommandCompletionService`. Or did you plan to put all the tests that are command related in this file?

File: azure-pipelines.yml (hunk 1 above)
Comment (stefanobaghino-da) [reply azure-pipelines.yml]: [Urban diction entry on dQw4w9WgXcQ](https://www.urbandictionary.com/define.php?term=dQw4w9WgXcQ)

There was a problem with CI, this served as a test, will soon rebase on top of master when #3323 is merged (which it probably already is).

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: Good to know I'm not the only clueless person reading this. I'll happily rephrase this, could you suggestions.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala (hunk 3 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: I'd keep it here for now and move it when we actually port the `CommandCompletionServiceIT` suite. Sounds good?

File: azure-pipelines.yml (hunk 1 above)
Comment (stefanobaghino-da) [reply azure-pipelines.yml]: Should be addressed by https://github.com/digital-asset/daml/pull/3320/commits/9e52f54a56b5569dc4f7f5898c1f6eb20f997d42

Will rebase on `master` if this is not enough.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: Kind of good news: this test is actually a duplicate of one found already in the `TransactionService` suite. I'll remove this one, but it's probably worth spending a couple minutes figuring out the meaning of the existing test.

Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: Addressed by https://github.com/digital-asset/daml/pull/3320/commits/979de36daf75e8b2382ae37d86ac694511badd63

Comment (gerolf-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: 👍

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala (hunk 3 above)
Comment (gerolf-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala]: Sure.

**Pull request merged**